rosutil/convenience.rkt

```diff
@@ -29,5 +32,11 @@
   ; so x%0 instead of x$0
   (@constant (string->uninterned-symbol (format "~a%~a" sym-base counter)) type))
 
+; instantiate a vector of fresh symbolics, in the same form as a Yosys memory
+(define (fresh-memory-like name mem)
+  (let ([elem-type (@type-of (vector-ref mem 0))])
+    (list->vector
+     (build-list (vector-length mem) (lambda (_) (fresh-symbolic name elem-type))))))
```

Comment:
I'm guessing this is based on similar code I had written somewhere else in the codebase? I'm not sure why I used `(list->vector (build-list ...))`, because we can directly just `build-vector` and avoid the extra conversion. I've fixed the other use cases, e.g. https://github.com/anishathalye/rtl/blob/master/yosys/yosys.rkt#L34

Reply:
Also, now that `fresh-symbolic` accepts strings/symbols, we could use a slightly more descriptive name here if we wanted. E.g. instead of just using the name of the field, like `ram`, we could build a string `(format "~a[~a]" name i)` (where i is the index of the vector), and then we'd have names like `ram[3]`, which might make it slightly nicer for debugging.

Reply:
Yup, this does come from code you wrote somewhere that I had kept around. I cleaned things up with `build-vector` and also used your better name suggestion (good idea!)